extendr-api/src/robj/into_robj.rs

```diff
@@ -297,6 +297,93 @@ macro_rules! impl_real_tvv {
 impl_real_tvv!(f64);
 impl_real_tvv!(f32);
 
+//TODO: It could be an idea to make this a attribute macro for users,
+// if they wish to specialise their own types as representable in R.
+
+/// Generates a [`ToVectorValue`] for a type, by inheriting the properties
+/// of another type's [`ToVectorValue`]'s implementation.
+///
+/// This is meant to be used for wrappers of types that may be represented
+/// in R. The marshalling rules for this is represented in `ToVectorValue`,
+/// and this macro merely co-opts
+///
+/// Arguments:
+///
+/// * `$type`         - Target type
+/// * `$synonym_type` - A type that has a `ToVectorValue`-impl, and an `inner`-method.
+///
+/// Requirements: `$type` must have an `inner`-method to extract the
+/// wrapped value. It suffices that `$type` implements `Scalar<T>`.
+///
+/// Example usage:
+///
+/// ```ignore
+/// impl_synonym_type(Rint, i32);
+/// ```
+///
+/// The example here implements
+///
+/// `impl ToVectorValue for Rint`,
+///
+/// this entails that `Rint` would be stored in `R` exactly
+/// as `i32`.
+///
+macro_rules! impl_synonym_type {
```

Comment:
`into_robj.rs` is like 1000 lines and basically impossible to browse. Can we move this new bit to a submodule and just reference it here?